liquibase-core/src/main/java/liquibase/lockservice/AbstractLockService.java

```diff
@@ -0,0 +1,112 @@
+package liquibase.lockservice;
+
+import liquibase.Scope;
+import liquibase.configuration.GlobalConfiguration;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LockException;
+import liquibase.plugin.AbstractPlugin;
+
+import java.text.DateFormat;
+import java.util.Date;
+
+/**
+ * Base convenience class for {@link LockService} implementations
+ */
+public abstract class AbstractLockService extends AbstractPlugin implements LockService {
+
+    protected Database database;
+
+    private Long changeLogLockPollRate;
+    private Long changeLogLockRecheckTime;
+
+
+    /**
+     * Default implementation does nothing
+     */
+    @Override
+    public void init() throws DatabaseException {
+
+    }
+
+    @Override
+    public void setDatabase(Database database) {
+        this.database = database;
+    }
+
+    public Long getChangeLogLockWaitTime() {
+        if (changeLogLockPollRate != null) {
+            return changeLogLockPollRate;
+        }
+        return LiquibaseConfiguration.getInstance().getConfiguration(GlobalConfiguration.class)
+                .getDatabaseChangeLogLockWaitTime();
+    }
+
+    @Override
+    public void setChangeLogLockWaitTime(long changeLogLockWaitTime) {
+        this.changeLogLockPollRate = changeLogLockWaitTime;
+    }
+
+    public Long getChangeLogLockRecheckTime() {
+        if (changeLogLockRecheckTime != null) {
+            return changeLogLockRecheckTime;
+        }
+        return LiquibaseConfiguration.getInstance().getConfiguration(GlobalConfiguration.class)
+                .getDatabaseChangeLogLockPollRate();
+    }
+
+    @Override
+    public void setChangeLogLockRecheckTime(long changeLogLockRecheckTime) {
+        this.changeLogLockRecheckTime = changeLogLockRecheckTime;
+    }
+
+
+    /**
+     * Default implementation does nothing
+     */
+    @Override
+    public void reset() {
+    }
+
+    @Override
+    public void waitForLock() throws LockException {
+
+        boolean locked = false;
+        long timeToGiveUp = new Date().getTime() + (getChangeLogLockWaitTime() * 1000 * 60);
```

Comment:
it might be worthwhile switching to `Duration` types instead of numeric types for wait and recheck times.